For some reasons that I didn't bother researching, sometimes
builds have one of the times in the list being None.
This becomes the minimum, gets sorted first, and this build
disrupts the grid view. As a hackish workaround, set the time
to now if it's None.

diff -ru -F '^[[:space:]]*\(def\|class\)[[:space:]]\+' -x '*.pyc' ./status/web/grid.py /usr/lib/python2.7/dist-packages/buildbot/status/web/grid.py
--- ./status/web/grid.py	2013-08-23 01:43:56.000000000 +0300
+++ /usr/lib/python2.7/dist-packages/buildbot/status/web/grid.py	2015-02-25 12:53:39.130293380 +0200
@@ -13,6 +13,7 @@
 #
 # Copyright Buildbot Team Members
 
+import time
 from twisted.internet import defer
 from buildbot.status.web.base import HtmlResource
 from buildbot.status.web.base import build_get_class, path_to_builder, path_to_build
@@ -152,7 +153,7 @@ def getRecentSourcestamps(self, status,
             for build in self.getRecentBuilds(builder, numBuilds, branch):
                 ss = build.getSourceStamps(absolute=True)
                 key = self.getSourceStampKey(ss)
-                start = min(build.getTimes())
+                start = min(build.getTimes()) or time.time()
                 if key not in sourcestamps or sourcestamps[key][1] > start:
                     sourcestamps[key] = (ss, start)
 
